File: 01-Arrays-and-Hashing/3. Longest Substring Without Repeating Characters/3. Longest Substring Without Repeating Characters.java
1class Solution {
2    public int lengthOfLongestSubstring(String s) {
3        int n = s.length();
4        int ans = 0;
5
6        for (int i = 0; i < n; i++) {
7            boolean[] visited = new boolean[256];
8            for (int j = i; j < n; j++) {
9                if (visited[s.charAt(j)]) {
10                    break;
11                }
12
13                visited[s.charAt(j)] = true;
14                ans = Math.max(ans, (j - i + 1));
15            }
16        }
17
18        return ans;
19    }
20}

File: 01-Arrays-and-Hashing/3. Longest Substring Without Repeating Characters/3. Longest Substring Without Repeating Characters.cpp
1class Solution {
2public:
3    int lengthOfLongestSubstring(string s) {
4        int n = s.length();
5        int ans = 0;
6
7        for (int i = 0; i < n; i++) {
8            vector<bool> visited(256, false);
9            for (int j = i; j < n; j++) {
10                if (visited[s[j]])
11                    break;
12                visited[s[j]] = true;
13                ans = max(ans, j - i + 1);
14            }
15        }
16        return ans;
17    }
18};